src/coreclr/jit/hwintrinsicarm64.cpp

```diff
@@ -652,6 +652,189 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
+            }
+
+            op1 = gtNewSimdHWIntrinsicNode(simdType, op1, op2, NI_AdvSimd_And, simdBaseJitType, simdSize,
+                                           /* isSimdAsHWIntrinsic */ false);
+
+            op1 = gtNewSimdHWIntrinsicNode(simdType, op1, op3, NI_AdvSimd_ShiftLogical, simdBaseJitType, simdSize,
+                                           /* isSimdAsHWIntrinsic */ false);
+
+            if (varTypeIsByte(simdBaseType) && (simdSize == 16))
+            {
+                CORINFO_CLASS_HANDLE clsHnd = gtGetStructHandleForSIMD(simdType, simdBaseJitType);
+
+                op1 = impCloneExpr(op1, &op2, clsHnd, (unsigned)CHECK_SPILL_ALL,
+                                   nullptr DEBUGARG("Clone op1 for vector extractmostsignificantbits"));
+
+                op1 = gtNewSimdHWIntrinsicNode(TYP_SIMD8, op1, NI_Vector128_GetLower, simdBaseJitType, simdSize,
+                                               /* isSimdAsHWIntrinsic */ false);
+                op1 = gtNewSimdHWIntrinsicNode(TYP_SIMD8, op1, NI_AdvSimd_Arm64_AddAcross, simdBaseJitType, 8,
+                                               /* isSimdAsHWIntrinsic */ false);
+                op1 = gtNewSimdHWIntrinsicNode(simdBaseType, op1, NI_Vector64_ToScalar, simdBaseJitType, 8,
+                                               /* isSimdAsHWIntrinsic */ false);
+                op1 = gtNewCastNode(TYP_INT, op1, /* isUnsigned */ true, simdBaseType);
+
+                GenTree* zero  = gtNewSimdHWIntrinsicNode(retType, NI_Vector128_get_Zero, simdBaseJitType, simdSize);
+                ssize_t  index = 8 / genTypeSize(simdBaseType);
+
+                op2 = gtNewSimdHWIntrinsicNode(simdType, op2, zero, gtNewIconNode(index), NI_AdvSimd_ExtractVector128,
+                                               simdBaseJitType, simdSize, /* isSimdAsHWIntrinsic */ false);
+                op2 = gtNewSimdHWIntrinsicNode(TYP_SIMD8, op2, NI_Vector128_GetLower, simdBaseJitType, simdSize,
+                                               /* isSimdAsHWIntrinsic */ false);
+                op2 = gtNewSimdHWIntrinsicNode(TYP_SIMD8, op2, NI_AdvSimd_Arm64_AddAcross, simdBaseJitType, 8,
+                                               /* isSimdAsHWIntrinsic */ false);
+                op2 = gtNewSimdHWIntrinsicNode(simdBaseType, op2, NI_Vector64_ToScalar, simdBaseJitType, 8,
+                                               /* isSimdAsHWIntrinsic */ false);
+                op2 = gtNewCastNode(TYP_INT, op2, /* isUnsigned */ true, simdBaseType);
+
+                op2     = gtNewOperNode(GT_LSH, TYP_INT, op2, gtNewIconNode(8));
+                retNode = gtNewOperNode(GT_OR, TYP_INT, op1, op2);
+            }
+            else
+            {
+                if (!varTypeIsLong(simdBaseType))
+                {
+                    op1 = gtNewSimdHWIntrinsicNode(TYP_SIMD8, op1, NI_AdvSimd_Arm64_AddAcross, simdBaseJitType,
```

Comment:
Same comment regarding using `AddAcross` versus many `AddPairwise`

Reply:
This is roughly the same scenario as https://github.com/dotnet/runtime/pull/63414#discussion_r783162088. We need a proper sum here that fits in `32-bits`.

There are some other potential optimization opportunities here, but they would require handling in lowering and aren't "critical" to getting the feature out, so it's likely better to handle them in a separate PR.
- These would involve specially handling the input and whether or not its an intrinsic that is known to produce an all-bits set vs no-bits set per-element result. In that scenario, we can elide the `shift` in favor of masking the right bits directly and then use two `AddPairwise`, which may be better.
- The same applies to certain codegen for `x86/x64` (but not `ExtractMostSignificantBits`, since that's just `MoveMask` there) where we could optimize to just things like `Blend` instead